src/control-plane/backend/lambda/api/service/application.ts

```diff
@@ -269,39 +271,41 @@ export class ApplicationServ {
           streamEnableAppIds.splice(index, 1);
         }
       }
-      const streamingSinkKinesisConfig = this.getStreamingSinkKinesisConfig(latestPipeline, streamEnableAppIds);
-      const updateRes = await updateFlinkApplicationEnvironmentProperties(latestPipeline.region, flinkAppName, streamingSinkKinesisConfig);
+
+      const streamingSinkKinesisConfig = await this.getStreamingSinkKinesisConfig(latestPipeline, streamEnableAppIds);
+      const updateRes = await updateFlinkApplication(latestPipeline.region, flinkAppName, streamEnableAppIds, streamingSinkKinesisConfig);
       if (!updateRes) {
         return res.status(500).json(new ApiFail('Failed to update Flink application environment properties.'));
       }
       latestPipeline.streaming = {
         ...latestPipeline.streaming,
         appIdStreamList: streamEnableAppIds,
+        retentionHours: latestPipeline.streaming?.retentionHours ?? 1,
       };
-      await store.updatePipelineAtCurrentVersion(latestPipeline);
+      const pipeline = new CPipeline(latestPipeline);
+      await pipeline.updateStreamingApp(streamEnableAppIds);
       return res.status(200).json(new ApiSuccess(null, 'Application streaming updated.'));
     } catch (error) {
       next(error);
     }
   };
 
-  private getStreamingSinkKinesisConfig(pipeline: IPipeline, streamingSinkKinesis: string[]) {
+  private async getStreamingSinkKinesisConfig(pipeline: IPipeline, streamEnableAppIds: string[]) {
     const enableConfigs: any[] = [];
     try {
-      const sinkKinesis = getStackOutputFromPipelineStatus(
-        pipeline.stackDetails ?? pipeline.status?.stackDetails,
-        PipelineStackType.STREAMING, OUTPUT_STREAMING_INGESTION_SINK_KINESIS_JSON);
-      if (!sinkKinesis) {
+      const streamingStack = await describeStack(
+        pipeline.region,
+        getStackName(pipeline.pipelineId, PipelineStackType.STREAMING, pipeline.ingestionServer.sinkType),
+      );
+      if (!streamingStack) {
         return enableConfigs;
       }
-      const sinks = JSON.parse(sinkKinesis);
-      for (let [key, value] of Object.entries(sinks)) {
-        if (streamingSinkKinesis.includes(key)) {
-          enableConfigs.push({
-            appId: key,
-            streamArn: value,
-          });
-        }
+      const streamingStackIdPrefix = streamingStack?.StackId?.split('/')[2].split('-')[0];
+      for (let appId of streamEnableAppIds) {
+        enableConfigs.push({
+          appId: appId,
+          streamArn: `arn:${awsPartition}:kinesis:${pipeline.region}:${awsAccountId}:stream/${SINK_STREAM_NAME_PREFIX}${pipeline.projectId}_${appId}_${streamingStackIdPrefix}`,
```

Comment:
why not using the output `SinkKinesisJson` in Stream stack?

Reply:
Updating the streaming stack while also updating the Flink configuration, unable to wait for output.